packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -138,11 +138,11 @@ class _GameWidgetState<T extends Game> extends State<GameWidget<T>> {
 
   MouseCursor? _mouseCursor;
 
-  Future<void> get _loaderFuture {
+  late final Future<void> _loaderFuture = (() {
```

Comment:
I don't think this will work, now it will only run `onMount` once, at the same time as `onLoad`?

Reply:
Actually it is off because it will not run onmount at all. I am not sure why we should run onmount every time the widget tree rebuilds?

Reply:
Whatever is the logic between onload and unmount, we should not generate a new future every time the build method is called, since a widget rebuild may happen even because of things not related to Flame. [In the previous implementation](https://github.com/flame-engine/flame/pull/906/files#diff-40f66f3222b3010300352eac3ec49ca49eb9f08a3148cf6e224655cfaa63aff4L195), we cached the future.

Reply:
Bottom line:: since `Game` is now a component, this seems more like a component lifecycle and should be treated as such: inside Flame. Hooks to the widget lifecycle may, be more specialized (on widget mount, on widget rebuild etc).

Reply:
I agree, it shouldn't trigger on each widget rebuild. But, we need `onMount` to trigger if the `GameWidget` is moved from one place in the tree to another (but is already loaded), or if it is removed and then later added back to the tree, is this even possible to differentiate from a normal widget rebuild?

Reply:
There are some possibilities here:

- The package user removes `GameWidget` from the tree temporarily: this unmounts the widget and kills the state (unless there is a global key keeping it), de-caching the future.
- The package user moves `GameWidget` in the widget tree: Same as above.
- The package user changes the game in which it passes to the `GameWidget`: The future is de-cached [here](https://github.com/flame-engine/flame/pull/1093/files#diff-40f66f3222b3010300352eac3ec49ca49eb9f08a3148cf6e224655cfaa63aff4R203).

If the user keeps the state in  global key, we should assume it wants to keep the state/

Reply:
Alright, seems like this is the way to go then. Are the lifecycle docs in line with this?